tests/app/main/views/test_index_page.py

```diff
@@ -120,107 +94,202 @@ def test_check_service_edits_link_is_shown_to_users_with_the_right_role(self, ro
+        ('standstill', True, True, True, True),
+        ('live', True, True, True, True),
+        ('expired', False, False, False, False),
+    ])
+    def test_framework_action_lists_only_contain_actions_for_framework_status(
+            self, framework_status, agreements_shown, stats_shown, comms_shown, contact_shown
+    ):
+        data_api_client.find_frameworks = mock.Mock()
+        data_api_client.find_frameworks.return_value = self._get_mock_framework_response(framework_status)
+        self.user_role = "admin-framework-manager"
+        response = self.client.get('/admin')
+        document = html.fromstring(response.get_data(as_text=True))
+
+        assert bool(document.xpath('.//a[contains(text(),"agreements")]')) == agreements_shown
+        assert bool(document.xpath('.//a[contains(text(),"View application statistics")]')) == stats_shown
+        assert bool(document.xpath('.//a[contains(text(),"Upload communications")]')) == comms_shown
+        assert bool(document.xpath('.//a[contains(text(),"Contact suppliers")]')) == contact_shown
+
+    @pytest.mark.parametrize("role, link_should_be_visible, expected_link_text", [
+        ("admin", False, None),
+        ("admin-ccs-category", True, "View agreements"),
+        ("admin-ccs-sourcing", True, "Countersign agreements"),
+        ("admin-framework-manager", True, "View agreements"),
+        ("admin-manager", False, None),
+    ])
+    def test_framework_action_list_includes_agreements_link(self, role, link_should_be_visible, expected_link_text):
+        self.user_role = role
+        response = self.client.get('/admin')
+        document = html.fromstring(response.get_data(as_text=True))
+        link_is_visible = bool(document.xpath('.//a[contains(text(),"agreements")]'))
+
+        assert link_is_visible is link_should_be_visible, (
+            "Role {} {} see the link".format(role, "can not" if link_should_be_visible else "can")
+        )
+        if expected_link_text:
+            assert document.xpath('.//a[contains(text(),"agreements")]//text()')[0] == expected_link_text, (
+                "Agreements link text for role {} is {}".format(role, expected_link_text)
+            )
+
     @pytest.mark.parametrize("role, link_should_be_visible", [
         ("admin", False),
         ("admin-ccs-category", False),
         ("admin-ccs-sourcing", False),
         ("admin-framework-manager", True),
         ("admin-manager", False),
     ])
-    def test_download_user_lists_link_is_shown_to_users_with_the_right_role(self, role, link_should_be_visible):
+    def test_framework_action_list_includes_contact_suppliers(self, role, link_should_be_visible):
         self.user_role = role
         response = self.client.get('/admin')
         document = html.fromstring(response.get_data(as_text=True))
-        link_is_visible = bool(document.xpath('.//a[@href="/admin/users/download"]'))
+        link_is_visible = bool(document.xpath('.//a[contains(text(),"Contact suppliers")]'))
```

Comment:
This is another one that feels like `data_api_client.find_frameworks.return_value` should be set before it will pass?  (Also for the next two)?

Reply:
Same answer as above :)